src/kmergraph.cpp

```diff
@@ -238,20 +218,271 @@ void KmerGraph::discover_k() {
     }
 }
 
-void KmerGraph::set_exp_depth_covg(const uint32_t edp) {
+std::ostream &operator<<(std::ostream &out, KmerGraph const &data) {
+    for (const auto &c: data.nodes) {
+        out << *(c) << std::endl;
+    }
+    return out;
+}
+
+//save the KmerGraph as gfa
+void KmerGraph::save(const std::string &filepath, const std::shared_ptr<LocalPRG> localprg) {
```

Comment:
Again, we should be moving towards standardising the use of `boost::filesystem` for working with paths. The API is extremely similar and you shouldn't need to change anything in the logic of the code.

Reply:
I think it is better to address all such cases when solving #68 and #139